natpmp.go

```diff
@@ -20,17 +21,23 @@ const RECOMMENDED_MAPPING_LIFETIME_SECONDS = 3600
 
 // Interface used to make remote procedure calls.
 type caller interface {
-	call(msg []byte) (result []byte, err error)
+	call(msg []byte, timeout time.Duration) (result []byte, err error)
 }
 
 // Client is a NAT-PMP protocol client.
 type Client struct {
-	caller caller
+	caller  caller
+	timeout time.Duration
 }
 
 // Create a NAT-PMP client for the NAT-PMP server at the gateway.
 func NewClient(gateway net.IP) (nat *Client) {
-	return &Client{&network{gateway}}
+	return &Client{&network{gateway}, 0}
+}
+
+// Create a NAT-PMP client for the NAT-PMP server at the gateway, with a timeout.
```

Comment:
Could you add more text to this comment, explaining what the timeout does, and how it compares to the default NewClient() behavior? (I can kind of figure it out from reading the implementation, but I think the comment should explain it well enough that the caller doesn't have to read the implementation.)